pkg/apis/eks.cattle.io/v1/types.go

```diff
@@ -52,26 +52,40 @@ type EKSClusterConfigSpec struct {
 }
 
 type EKSClusterConfigStatus struct {
-	Phase          string   `json:"phase"`
-	VirtualNetwork string   `json:"virtualNetwork"`
-	Subnets        []string `json:"subnets"`
-	SecurityGroups []string `json:"securityGroups"`
+	Phase                    string    `json:"phase"`
+	VirtualNetwork           string    `json:"virtualNetwork"`
+	Subnets                  []string  `json:"subnets"`
+	SecurityGroups           []string  `json:"securityGroups"`
+	ManagedLaunchTemplateID  string    `json:"managedLaunchTemplateID"`
+	TemplateVersionsToDelete []*string `json:"templateVersionsToDelete"`
 	// describes how the above network fields were provided. Valid values are provided and generated
 	NetworkFieldsSource string `json:"networkFieldsSource"`
 	FailureMessage      string `json:"failureMessage"`
 }
 
 type NodeGroup struct {
-	Gpu           *bool               `json:"gpu"`
-	NodegroupName *string             `json:"nodegroupName" norman:"required"`
-	DiskSize      *int64              `json:"diskSize"`
-	InstanceType  *string             `json:"instanceType" norman:"required"`
-	Labels        map[string]*string `json:"labels"`
-	Ec2SshKey     *string             `json:"ec2SshKey"`
-	DesiredSize   *int64              `json:"desiredSize"`
-	MaxSize       *int64              `json:"maxSize"`
-	MinSize       *int64              `json:"minSize"`
-	Subnets       []string           `json:"subnets"`
-	Tags          map[string]*string `json:"tags"`
-	Version       *string             `json:"version"`
+	Gpu                  *bool              `json:"gpu"`
+	ImageID              *string            `json:"imageId"`
+	NodegroupName        *string            `json:"nodegroupName" norman:"required"`
+	DiskSize             *int64             `json:"diskSize"`
+	InstanceType         *string            `json:"instanceType" norman:"required"`
+	Labels               map[string]*string `json:"labels"`
+	Ec2SshKey            *string            `json:"ec2SshKey"`
+	DesiredSize          *int64             `json:"desiredSize"`
+	MaxSize              *int64             `json:"maxSize"`
+	MinSize              *int64             `json:"minSize"`
+	Subnets              []string           `json:"subnets"`
+	Tags                 map[string]*string `json:"tags"`
+	ResourceTags         map[string]*string `json:"resourceTags"`
+	UserData             *string            `json:"userData"`
+	Version              *string            `json:"version"`
+	LaunchTemplate       *LaunchTemplate    `json:"launchTemplate"`
+	RequestSpotInstances *bool              `json:"requestSpotInstances"`
+	SpotInstanceTypes    []*string          `json:"spotInstanceTypes"`
```

Comment:
I think for subnets I did []string because slices I wouldn't have to consider nil elements, do you have thoughts on one over the other?

Reply:
My guess is that these probably wouldn't be nil from syncing, just someone sending it that way. This is good because it probably matches AWS API. As long as someone sending nil elements doesn't cause a panic or anything this should be good.

Reply:
Yes, the reason I made `SpotInstanceTypes` a `[]*string` was to match the AWS API. I verified that passing a `nil` value will not panic. However, passing a value of `[]*string{nil}` will fail to create the node group, but I would classify that as a user error.